R/module_teal.R

```diff
@@ -160,18 +160,19 @@ srv_teal <- function(id, modules, teal_data_rv, filter = teal_slices()) {
       )
     }
 
-    env <- environment()
     datasets_reactive <- eventReactive(teal_data_rv(), {
-      env$progress <- shiny::Progress$new(session)
-      env$progress$set(0.25, message = "Setting data")
-
+      progress_d <- Progress$new(
+        max = length(unlist(module_labels(modules)))
+      )
+      progress_d$set(message = "Preparing data filtering")
+      on.exit(progress_d$close())
```

Comment:
maybe instead `on.exit` go with `shiny::onStop`?

Reply:
No, you want the bar do disappear when this funciton exits, not when `shiny` stops.

Reply:
got it! ignore me on this one